api/handlers.go

```diff
@@ -341,6 +341,30 @@ func getVolumes(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	danglingOnly := false
+	refVolumes := make(map[string]bool)
+
+	if filters.Include("dangling") {
+		if filters.ExactMatch("dangling", "true") {
+			danglingOnly = true
+			for _, container := range c.cluster.Containers() {
+				for _, mount := range container.Mounts {
+					if mount.Type == "volume" {
+						if mount.Driver == "local" {
```

Comment:
Does this mean that the `dangling` filter will not work for non-local volumes?

Reply:
no, it still works for all, local and non-local. 
The condition on line 353 is there to honor the "existing" naming convention used around line 417 for local volumes

https://github.com/dani-docker/swarm/blob/9ae1c6e16226bceae46afb551a3ed3287c8d6176/api/handlers.go#L405-L418